### ISSUE TITLE
Details of the incentive/punishment mechanism in staker contracts

### ISSUE THREAD

**Issue by jtse0** [SolomonDefi/solomon-monorepo#123]
Details relating to the incentive and punishment mechanisms related to staker contracts need to be clarified and added to the staker contracts.

Currently, there are monitors in place to track total outstanding vote obligations and voting obligations per dispute. We can go with the model of allowing a certain grace allowance per user and if they go over that grace allowance, then as a punishment they don't get rewards.  Will there be further punishments that need to be implemented as well?

In terms of rewards, how will rewards be calculated and distributed to these jurors/stakers?

Notes:
**Outstanding votes** are for jurors - it is the count of dispute votes that they have to complete, when they are selected as a juror for a dispute vote it is +1 to their outstanding vote obligation

**Grace allowance** is how many votes they can miss and still be able to get rewarded for fulfilling their juror requirements

**Comment by jtse0** [comment]
After discussion, there are few points that need to be clarified and expanded upon.

[To be confirmed]
There will be a basic reward that will be for general stakers on a periodic basis.  There shall be an extra juror reward every time disputes votes occur.

There still needs to be further discussion on how the non-voters or minority voters are "punished." It seems as though for a majority of cases, punishing the voters who end up with the minority votes will help ensure that the system works as intended and votes are made after a careful evaluation of the evidence.  As per discussion, making sure that votes are encrypted will help reduce the possibility of collusion among jurors which might skew results.  Also, as an additional thought, perhaps there can be a mechanism in place to allow jurors to make a claim to get their rewards in the case of close votes, or simply reward everyone if the ends up being something like a 40/60 or 50/50 close vote.

[Items that can proceed]
The withdrawal timer for stakers will simply record the date of their previous withdrawal and set a default wait period that every staker must wait before they can withdraw their rewards.  As long as the default period has passed, stakers are are able to withdraw their rewards at any moment.  This might introduce different ways to game the system, but these ways of manipulating the way rewards are received are acceptable and don't post any significant risks to the system running as intended.

**Comment by solomondefi-dev** [comment]
Sounds good, I will follow up on the two discussion points, then we can create new issues and close this one:

1. Confirm details (amount/formula) of the periodic reward for general stakers. To be implemented after #144 (it can probably be included in #145 rather than creating a new issue).
2. Confirm details (punishment strategy and reward amount formula) of the extra juror reward for voting on disputes.

**Comment by solomondefi-dev** [comment]
Covered in other issues, will make a new issue if needed after review.
